wasm-commp/src/lib.rs

```diff
@@ -113,15 +116,124 @@ pub fn calculate_piece_commitment<R: Read>(
     let mut buffer = [0; NODE_SIZE];
     let num_leafs = piece_size.div_ceil(NODE_SIZE as u64) as usize;
 
-    let leaves = (0..num_leafs)
-        .map(|_| {
-            fr32_reader.read_exact(&mut buffer).unwrap();
-            buffer
-        })
-        .collect::<Vec<[u8; 32]>>();
+    // Use a `for` loop instead of `.map()` so we can use the `?` operator
+    // for proper error propagation when reading from the Fr32Reader.
+    let mut leaves = Vec::with_capacity(num_leafs);
+    for _ in 0..num_leafs {
+        fr32_reader
+            .read_exact(&mut buffer)
+            .map_err(|e| JsValue::from_str(&format!("Read error: {}", e)))?;
+        leaves.push(buffer);
+    }
```

Comment:
You can also do it like this. Possible because of the [impl](https://doc.rust-lang.org/std/result/enum.Result.html#impl-FromIterator%3CResult%3CA,+E%3E%3E-for-Result%3CV,+E%3E)

```suggestion
    let leaves = (0..num_leafs)
        .map(|_| {
            fr32_reader
                .read_exact(&mut buffer)
                .map_err(|e| JsValue::from_str(&format!("Read error: {}", e)))?;
            Ok(buffer)
        })
        .collect::<Result<Vec<_>, JsValue>>()?;
```